### PR TITLE
[SPARK-24899][SQL][DOC] Add example of monotonically_increasing_id standard function to scaladoc

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

Example of `monotonically_increasing_id` standard function (with how it works internally) in scaladoc

## How was this patch tested?

Local build. Waiting for Jenkins

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -1150,16 +1150,48 @@ object functions {
...
-   * This expression would return the following IDs:
-   *
    * {{{
-   * 0, 1, 2, 8589934592 (1L << 33), 8589934593, 8589934594.
+   * // Create a dataset with four partitions, each with two rows.
+   * val q = spark.range(start = 0, end = 8, step = 1, numPartitions = 4)
+   *
+   * // Make sure that every partition has the same number of rows
+   * q.mapPartitions(rows => Iterator(rows.size)).foreachPartition(rows => assert(rows.next == 2))
+   * q.select(monotonically_increasing_id).show
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: eh @jaceklaskowski, wouldn't this one be enough as an example?

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: I know you're exploring the internals but .. to be honest I was wondering if users are usually interested in such in-deep explanation since I guess most of them wouldn't care about the details.

Comment (TomaszGaweda) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: IMHO It' enough to add that rows are consecutive in each partition, but not between partitions and that values are shifted left by 33 - written in words, not code, will be much shorter and concise

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala [hunk 2]
```diff
@@ -80,7 +80,5 @@ case class MonotonicallyIncreasingID() extends LeafExpression with Stateful {
 
   override def prettyName: String = "monotonically_increasing_id"
 
-  override def sql: String = s"$prettyName()"
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala]: Why this change?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 1 above)
Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: I personally would simplify the example to not focus on the particular shift; yeah that behavior ought not change but it's not really something a caller would ever rely on. And I think you don't need to make a new variable to subtract 1 from row number, etc. Something simply showing the two properties -- increasing within partition, not between partitions -- is enough.

**Review by srowen — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala (hunk 2 above)
Comment (jaceklaskowski) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala]: It's the default and no need for the override, isn't it?

**Comment by srowen** [comment]
I think this is mergeable if the examples are simplified a bit per comments above @jaceklaskowski

**Comment by srowen** [comment]
@jaceklaskowski would you like to update this?

**Comment by srowen** [comment]
Ping @jaceklaskowski to update or close
